### ISSUE TITLE
Blitz 3.11.0 not working with CraftCMS 3.7.2x

### ISSUE THREAD

**Issue by cesalazar** [putyourlightson/craft-blitz#358]
### Describe the bug
Blitz doesn't seem to be able to generate a cache after upgrading:
- CraftCMS 3.6.17 => 3.7.21 (or 3.7.23)
- Blitz 3.10.0 => 3.11.0

We didn't change any config setting, clicked the Warm Cache button, and used the command `craft blitz/cache/refresh` in the console, in each case we got the same outcome: no cache generated.

We set `'debug' => true` in `config/blitz.php` and executed the command again; the log content is identical with and without `debug`. This is the console output:

```bash
$ ./craft blitz/cache/refresh
Blitz cache successfully cleared.
Blitz cache successfully flushed.
Cache purging is disabled.
Warming Blitz cache...
[>                                                                                                                    ] 0% (0/143) ETA: n/a
Warmed 0 of 143 pages. To see why pages were not cached, enable the `debug` config setting and then open the `storage/logs/blitz.log` file.
Blitz cache warming complete.
Deploying is disabled.

$ tail storage/logs/blitz.log
...
2021-12-01 16:47:16 [][][debug] Page not cached because it does not match an included URI pattern. [/en/redacted]
2021-12-01 16:47:16 [][][info] Blitz cache successfully cleared.
2021-12-01 16:47:17 [][][info] Blitz cache successfully flushed.
2021-12-01 16:47:17 [][][info] Blitz cache warming complete.
```

Then we switched the Cache Storage from Yii to Blitz File Storage, and navigated to `cpTrigger/utilities/blitz-cache`, where we can see a table with zero Cached Pages (see screenshot).

Then we switched the Cache Warmer from Guzzle to Local warmer, same result: no cache generated.

All these steps were tested both in the CP and the console using the command `craft blitz/cache/refresh`, and downgrading Blitz to version 3.10.0, 3.9.0, and 3.8.0.

We also read through the tickets #123 and #295, and tested with and without config file (our preferred usage). Also disabled the [async-queue](https://github.com/ostark/craft-async-queue) plugin, which was mentioned in one of the closed issues we came across.

When we clear all other caches (e.g. `craft clear-caches/all` and `craft cache/flush-all`), Blitz appears briefly in Craft's Queue in the CP. Making any changes in the code of the templates is reflected right away after a page refresh.

### To reproduce
Steps to reproduce the behaviour in the browser:
1. Update CraftCMS to v3.7.2x
2. Navigate to `cpTrigger/utilities/blitz-cache`
3. Click the Refresh Cache button

Steps to reproduce the behaviour in the console:
1. Update CraftCMS to v3.7.2x
2. SSH into the instance and `cd` into Craft's folder
3. Execute `craft blitz/cache/refresh`

### Expected behaviour
Blitz should generate the cache in the CP (automatically or manually) and in the console.

### Screenshots
- Zero Cached Pages using Blitz File Storage:
![blitz-no-cached-pages](https://user-images.githubusercontent.com/805258/144269243-4918a82c-961d-4ed0-8c02-b728f35aa98d.png)

- Output of executing `craft blitz/cache/refresh` with and without `debug` set:
![blitz-cmd-error](https://user-images.githubusercontent.com/805258/144271892-ccf4e231-ae55-4d42-894c-0b1f5528c51d.png)

### Versions
- Plugin version: 3.11.0
- Craft version: 3.7.21 or 3.7.23

**Comment by cesalazar** [comment]
I was able to solve this issue: all of our Sites had just an absolute path (e.g `\de`) as Base URL. Setting the URL using the alias `@baseUrl` followed by the language slug solved the problem. As mentioned in the ticket, Blitz was working fine with our config before the upgrade.

Funny enough, I found this thanks to Sherlock complaining about not being able to resolve the Sites' Base URL when I tried to switch the site using the dropdown control. It would be nice if Blitz error'd in a similar way.

![sherlock-base-url-not-found](https://user-images.githubusercontent.com/805258/144289647-08ff3202-396a-4c06-8c7c-bd11f4571c71.png)
